storage/storage_reference.go

```diff
@@ -30,6 +30,16 @@ func newReference(transport storageTransport, named reference.Named, id string)
 	if named == nil && id == "" {
 		return nil, ErrInvalidReference
 	}
+	if named != nil {
+		if reference.IsNameOnly(named) {
+			return nil, errors.Wrapf(ErrInvalidReference, "reference %s has neither a tag nor a digest", named.String())
+		}
+	}
+	if id != "" {
```

Comment:
Same comment as above.

Reply:
It’s not possible to do `if (expression) && newVariable := …; expression(newVariable)`.